### PR TITLE
[AMP OP&Test]numel op fp/bf 16 support

### PR DESCRIPTION
**Pull request by piDack**
### PR types
New features

### PR changes
Others

### Describe
numel op fp/bf 16 support

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_numel_op.py [hunk 1]
```diff
@@ -37,16 +38,72 @@ def test_check_output(self):
...
+        x = np.random.random((self.shape)).astype(self.dtype)
+        self.inputs = {'Input': convert_float_to_uint16(x)}
+        self.outputs = {'Out': np.array([np.size(x)])}
+
+    def test_check_output(self):
+        place = paddle.CUDAPlace(0)
+        self.check_output_with_place(place)
+
+    def init(self):
+        self.dtype = np.float32
```
Comment (ZzSean) [inline python/paddle/fluid/tests/unittests/test_numel_op.py]: dtype需要设置成uint16

Comment (piDack) [reply python/paddle/fluid/tests/unittests/test_numel_op.py]: 这个地方通过convert_float_to_uint16会转成uint16的

**Review by ZzSean — DISMISSED** [review]
LGTM

**Review by ZzSean — APPROVED** [review]
LGTM

**Review by qili93 — APPROVED** [review]
LGTM

**Pull request merged**